### PR TITLE
Add libarrow to AzureLinux

### PR DESCRIPTION
**Pull request by himaja-kesari**
libarrow is needed for ceph, adding a new package from fedora.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge
 

Buddy Build: 
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=533431&view=results

### CODE REVIEW

File: SPECS/libarrow/libarrow.spec [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+%files -n parquet-libs-devel
+%dir %{_includedir}/parquet/
+     %{_includedir}/parquet/*
+%{_libdir}/cmake/Parquet/*.cmake
+%{_libdir}/libparquet.so
+%{_libdir}/pkgconfig/parquet*.pc
+ 
+%changelog
+* Fri Mar 22 2024 Himaja Kesari <himajakesari@microsoft.com> - 15.0.0-4
```
Comment (anphel31) [inline SPECS/libarrow/libarrow.spec:246]: Why not release 2? (one greater than the Fedora version)?

File: SPECS/libarrow/libarrow.spec [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+%bcond_with use_s3
+%bcond_without have_rapidjson
+%bcond_without have_re2
+%bcond_without have_utf8proc
+ 
+Name:	libarrow
+Version:	15.0.0
+Release:	4%{?dist}
+Summary:	A toolbox for accelerated data interchange and in-memory processing
+License:	Apache-2.0
```
Comment (anphel31) [inline SPECS/libarrow/libarrow.spec:18]: Can you please add the following to match our other specs:
Group:
Vendor:
Distribution:

**Review by anphel31 — COMMENTED** [review]
Can you please restore and fill out the PR Merge Checklist?

File: SPECS/libarrow/libarrow.spec (hunk 2 above)
Comment (himaja-kesari) [reply SPECS/libarrow/libarrow.spec:18]: added them

File: SPECS/libarrow/libarrow.spec (hunk 1 above)
Comment (himaja-kesari) [reply SPECS/libarrow/libarrow.spec:249]: added changelog till fedora release 3

**Review by anphel31 — APPROVED** [review]

**Pull request merged**